.github/workflows/install_firedrake_test.yml

```diff
@@ -0,0 +1,86 @@
+name: 'Install Firedrake and Mesh Adaptation software'
+
+on:
+  # Trigger when this or any file in the install directory changes
+  pull_request:
+    paths:
+      - 'install/**'
+      - '.github/workflows/install_firedrake_test.yml'
+
+  # Trigger the workflow every Saturday at 1AM
+  schedule:
+    - cron: '0 1 * * 6'
+
+  # Manually trigger the workflow
+  workflow_dispatch:
+
+# Stop the workflow if a new run is requested
+concurrency:
+  group: 'install_firedrake_test-${{ github.ref }}'
+  cancel-in-progress: true
+
+jobs:
+  install_firedrake:
+    name: 'Install Firedrake'
+    runs-on: ubuntu-latest
+    container:
+      image: ubuntu:24.04
+      options: --user root
+
+    steps:
+      - name: 'Install required packages'
+        run: |
+          apt-get update
+          apt-get -y dist-upgrade
+          apt-get install -y --no-install-recommends tzdata curl vim docker.io \
+          openssh-client build-essential autoconf automake cmake gfortran git \
+          libopenblas-serial-dev libtool python3-dev python3-pip python3-tk \
+          python3-venv python3-requests zlib1g-dev libboost-dev sudo gmsh bison flex \
+          ninja-build libocct-ocaf-dev libocct-data-exchange-dev swig graphviz \
+          libcurl4-openssl-dev libxml2-dev
+
+      - name: 'Download necessary files'
+        run: |
+          curl -O https://raw.githubusercontent.com/mesh-adaptation/mesh-adaptation-docs/fix-mpi-install/install/Makefile
+          curl -O https://raw.githubusercontent.com/mesh-adaptation/mesh-adaptation-docs/fix-mpi-install/install/petsc_configure_options.txt
+
+      - name: 'Install Firedrake'
+        run: |
+          yes | make install_no_chk
+          cd firedrake-*
+          echo "FIREDRAKE_ENV=$(pwd)" >> $GITHUB_ENV
+
+      - name: 'Print firedrake-install.log if installation fails'
+        if: ${{ failure() }}
+        run: |
+          sleep 10
```

Comment:
Why `sleep 10`?

Reply:
I added a step that prints `firedrake-install.log` in case the installation step fails, but it wouldn't print the last parts of the file (i.e., where the error occurs), so it was useless for debugging. I then added a tmate session in https://github.com/mesh-adaptation/docs/actions/runs/12261678921/job/34209233179 and manually did `cat firedrake-install.log` and those missing lines were indeed there. So I thought that maybe the CI jumps to the next step as soon as the error is hit. But I'm not sure if this `sleep 10` helps though (the installations after that succeeded)

Reply:
Hm seems odd. I guess it's okay for now, but would be better to store the logs somewhere. Needing to `sleep` is always concerning.

Reply:
In https://github.com/mesh-adaptation/docs/pull/77/commits/d4c5b29a5e933f3e25c1052ddf70b750f6f9ff87 I changed it to instead upload the log as an artifact (no matter if the installation succeeds or fails). This will give us a link to download it and inspect it if we want to (exactly as the automatically built docs). Does that sound better?

Reply:
Ooh nice! That's much better indeed.